sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -352,6 +352,12 @@ private void initializeHttpRequest(BlockStmt body, HttpRequestContext method) {
             body.addStatement(StaticJavaParser.parseStatement("String host = " + method.getHost() + ";"));
         }
 
+        // Iterate through the query parameters and append them to the host string if they are not null
+        method.getQueryParams().forEach((key, value) -> {
+            body.addStatement("if (" + value + " != null) {" + "    host = CoreUtils.appendQueryParam(host, \"" + key
```

Comment:
If we have the CoreUtils method to append query param, we can do the null check there itself instead of doing the null check here for every single query param.